avm/res/sql/server/database/backup-long-term-retention-policy/main.bicep

```diff
@@ -28,10 +34,12 @@ resource server 'Microsoft.Sql/servers@2023-08-01-preview' existing = {
   }
 }
 
-resource backupLongTermRetentionPolicy 'Microsoft.Sql/servers/databases/backupLongTermRetentionPolicies@2023-08-01-preview' = {
+resource backupLongTermRetentionPolicy 'Microsoft.Sql/servers/databases/backupLongTermRetentionPolicies@2023-05-01-preview' = {
```

Comment:
Because the latest API version has an issue, and does not recognize `backupStorageAccessTier` and `makeBackupsImmutable` properties
See the [2024-08-01 version](https://github.com/Azure/bicep-types-az/blob/main/generated/sql/microsoft.sql/2023-08-01-preview/types.md#properties-133) (wrong)
Vs the [2024-05-01 version](https://github.com/Azure/bicep-types-az/blob/main/generated/sql/microsoft.sql/2023-05-01-preview/types.md#longtermretentionpolicyproperties) (which is good)

Reply:
Good to know. 

For this kind of implementation decision it could be a good idea to write ADR as a comment.